components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -639,4 +659,42 @@ private String getServiceProviderName(String clientId, String tenantDomain) {
         }
         return spName;
     }
+
+    private void handleMissingSessionStateGracefully(HttpServletRequest request, HttpServletResponse response)
+            throws IOException {
+
+        String redirectURL = OIDCSessionManagementUtil.getOIDCLogoutURL();
+        String idTokenHint = request.getParameter(OIDCSessionConstants.OIDC_ID_TOKEN_HINT_PARAM);
+        String postLogoutRedirectUri = request.getParameter(OIDCSessionConstants.OIDC_POST_LOGOUT_REDIRECT_URI_PARAM);
+        if (StringUtils.isEmpty(idTokenHint) || StringUtils.isEmpty(postLogoutRedirectUri)) {
+            response.sendRedirect(getRedirectURL(redirectURL, request));
+            return;
+        }
+        if (!validateIdToken(idTokenHint)) {
+            String msg = "ID token signature validation failed.";
+            log.error(msg);
```

Comment:
This error log seems not useful. Can we add some context information to this to identify which validation failed? If we can't we may not need the log, as we already respond with the same error

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1130